### PR TITLE
Allow to get console log and tests from build

### PR DESCRIPTION
**Pull request by radtriste**

### CODE REVIEW

File: vars/util.groovy [hunk 1]
```diff
@@ -352,3 +352,44 @@ def rmPartialDeps(){
         sh "find . -regex \".*\\.part\\(\\.lock\\)?\" -exec rm -rf {} \\;"
     }
 }
+
+String retrieveConsoleLog(String buildUrl = "${BUILD_URL}", int numberOfLines = 750) {
+    sh "rm -rf consoleText && wget --no-check-certificate ${buildUrl}consoleText && tail -n ${numberOfLines} consoleText > console.log"
+    return sh(returnStdout: true, script: 'cat console.log')
```
Comment (Ginxo) [inline vars/util.groovy:358]:
```suggestion
    return sh(returnStdout: true, script: "wget --no-check-certificate -qO - ${buildUrl}consoleText | tail -n ${numberOfLines}")
```

File: vars/util.groovy [hunk 2]
```diff
@@ -352,3 +352,44 @@ def rmPartialDeps(){
         sh "find . -regex \".*\\.part\\(\\.lock\\)?\" -exec rm -rf {} \\;"
     }
 }
+
+String retrieveConsoleLog(String buildUrl = "${BUILD_URL}", int numberOfLines = 750) {
+    sh "rm -rf consoleText && wget --no-check-certificate ${buildUrl}consoleText && tail -n ${numberOfLines} consoleText > console.log"
```
Comment (Ginxo) [inline vars/util.groovy:357]:
```suggestion
```

File: vars/util.groovy [hunk 3]
```diff
@@ -352,3 +352,44 @@ def rmPartialDeps(){
...
+    return sh(returnStdout: true, script: 'cat console.log')
+}
+
+def retrieveTestResults(String buildUrl = "${BUILD_URL}") {
+    sh "rm -rf testResults.json && wget --no-check-certificate -O testResults.json ${buildUrl}testReport/api/json"
+    return readJSON(file: 'testResults.json')
```
Comment (Ginxo) [inline vars/util.groovy:363]:
```suggestion
    return readJSON(text: sh(returnStdout: true, script: "wget --no-check-certificate -qO - ${buildUrl}testReport/api/json"))
```

File: vars/util.groovy [hunk 4]
```diff
@@ -352,3 +352,44 @@ def rmPartialDeps(){
...
     }
 }
+
+String retrieveConsoleLog(String buildUrl = "${BUILD_URL}", int numberOfLines = 750) {
+    sh "rm -rf consoleText && wget --no-check-certificate ${buildUrl}consoleText && tail -n ${numberOfLines} consoleText > console.log"
+    return sh(returnStdout: true, script: 'cat console.log')
+}
+
+def retrieveTestResults(String buildUrl = "${BUILD_URL}") {
+    sh "rm -rf testResults.json && wget --no-check-certificate -O testResults.json ${buildUrl}testReport/api/json"
```
Comment (Ginxo) [inline vars/util.groovy:362]:
```suggestion
```

**Review by Ginxo — CHANGES_REQUESTED** [review]
can you please check my proposal?
tests should be adapted

**Comment by radtriste** [comment]
changes tested here https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/custom/job/tradisso/job/test/89/
with jenkinsfile
```
@Library('jenkins-pipeline-shared-libraries')_

pipeline {
    agent any
    stages {
        stage('Test') {
            steps {
                script {
                    buildUrl = 'https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/custom/job/tradisso/job/kogito-runtimes-deploy/11/'
                    
                    String consoleLog = retrieveConsoleLog(buildUrl)
                    echo consoleLog
                    
                    def testResults = retrieveTestResults(buildUrl)
	                echo "Passed: ${testResults.passCount}"
                    echo "Failed: ${testResults.failCount}"
                    
                    def testsFailed = retrieveFailedTests(buildUrl)
                    testsFailed.each { failedTest -> 
                    	echo failedTest.fullName
                    	echo failedTest.url
                    }
                    echo "Passed: ${testResults.passCount}"
                    echo "Failed: ${testResults.failCount}"
                }
            }
        }
    }
}
```

**Review by Ginxo — APPROVED** [review]
LGTM

**Pull request merged**